go/engine/kex2_provisionee.go

```diff
@@ -565,7 +565,7 @@ func (e *Kex2Provisionee) pushLKSServerHalf() error {
 		return err
 	}
 
-	err = libkb.PostDeviceLKS(e.G(), e, e.device.ID, e.device.Type, e.lks.GetServerHalf(), e.lks.Generation(), chrText, chrKID)
+	err = libkb.PostDeviceLKS(context.TODO(), e.G(), e, e.device.ID, e.device.Type, e.lks.GetServerHalf(), e.lks.Generation(), chrText, chrKID)
```

Comment:
can't this be ctx.GetNetContext() too?  will have to pass ctx into this func.

Reply:
OK, let me see if I can work that out.

Reply:
We'd have to stick it into the `Engine`, which might be OK, but just different...

Reply:
I thought it was in the engine context already

Reply:
The context is passed in via `Run`, so we'd have to loop it through....

Reply:
Actually probably the right thing to do is to change this interface:

```go
// Provisionee is an interface that abstracts out the crypto and session
// management that a provisionee needs to do as part of the protocol.
type Provisionee interface {
	GetLogFactory() rpc.LogFactory
	HandleHello(keybase1.HelloArg) (keybase1.HelloRes, error)
	HandleHello2(keybase1.Hello2Arg) (keybase1.Hello2Res, error)
	HandleDidCounterSign([]byte) error
	HandleDidCounterSign2(keybase1.DidCounterSign2Arg) error
}
```

to pass `context.Context` down.... But that's a pretty big change, so best to leave that for a different PR, no?  *Plus* it might actually change the behavior and introduce bugs...

Reply:
I.e.,:

```go
// DidCounterSign2 is called via the RPC server interface by the remote client.
// It in turn delegates the work to the passed in Provisionee interface,
// calling HandleDidCounterSign()
func (p *provisionee) DidCounterSign2(_ context.Context, arg keybase1.DidCounterSign2Arg) (err error) {
	p.startedCounterSign <- struct{}{}
	err = p.arg.Provisionee.HandleDidCounterSign2(arg)
	p.done <- err
	return err
}
```
Should change to pass the context through...

Reply:
This follow on task is here: https://keybase.atlassian.net/browse/CORE-6022

Reply:
Yes, different PR sounds good.